frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -421,6 +424,76 @@ final class BloopBspServices(
     Task.eval { originToCompileStores.remove(params.originId); () }.executeAsync
   }
 
+  def linkProjects(
+      userProjects: Seq[ProjectMapping],
+      state: State,
+      compileArgs: List[String],
+      originId: Option[String],
+      logger: BspServerLogger
+  ): BspResult[bsp.CompileResult] = {
+    import bloop.engine.tasks.LinkTask.{linkJS, linkNative}
+
+    def doLink(
+        project: Project,
+        newState: State
+    ) =
+      project.platform match {
+        case platform @ Js(config, _, _) =>
+          val cmd = Commands.Link(List(project.name))
+          val targetDir = ScalaJsToolchain.linkTargetFrom(project, config)
+          linkJS(cmd, project, newState, false, None, targetDir, platform).map { linkState =>
+            val result = if (linkState.status == ExitStatus.LinkingError) {
+              Right(bsp.CompileResult(originId, bsp.StatusCode.Error, None, None))
+            } else {
+              Right(bsp.CompileResult(originId, bsp.StatusCode.Ok, None, None))
+            }
+            (linkState, result)
+          }
+        case Jvm(_, _, _, _, _, _) => ???
```

Comment:
@tgodzik Should we error here or just do a NoOp? if we do a NoOp a client can request linking all projects and only the ones that can link will be linked and trying to link JVM doesn't stop the linking. If we error  the client has to be explicit and only try to link targets that can be linked.

Reply:
We should probably just compile normally in this case and ignore link option. A warning could be printed